app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt

```diff
@@ -0,0 +1,305 @@
+import org.mozilla.fenix.onboarding.interactor.DefaultOnboardingInteractor
+import org.mozilla.fenix.onboarding.interactor.OnboardingInteractor
+import org.mozilla.fenix.onboarding.view.OnboardingView
+import org.mozilla.fenix.settings.SupportUtils
+import org.mozilla.fenix.settings.deletebrowsingdata.deleteAndQuit
+import org.mozilla.fenix.theme.ThemeManager
+import org.mozilla.fenix.whatsnew.WhatsNew
+import java.lang.ref.WeakReference
+
+class OnboardingFragment : Fragment() {
+
+    private var _binding: FragmentOnboardingBinding? = null
+    private val binding get() = _binding!!
+
+    private lateinit var onboardingStore: OnboardingStore
+    private lateinit var interactor: OnboardingInteractor
+    private lateinit var onboardingAccountObserver: OnboardingAccountObserver
+    private lateinit var onboardingView: OnboardingView
+
+    private val store: BrowserStore
+        get() = requireComponents.core.store
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        _binding = FragmentOnboardingBinding.inflate(inflater, container, false)
+
+        onboardingStore = StoreProvider.get(this) {
+            OnboardingStore()
+        }
+
+        onboardingAccountObserver = OnboardingAccountObserver(
+            context = requireContext(),
+            dispatchChanges = { mode ->
+                if (mode != onboardingStore.state.state) {
+                    onboardingStore.dispatch(OnboardingFragmentAction.UpdateState(mode))
+                }
+            }
+        )
+
+        interactor = DefaultOnboardingInteractor()
+
+        onboardingView = OnboardingView(
+            containerView = binding.sessionControlRecyclerView,
+            interactor = interactor
+        )
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        observeSearchEngineChanges()
+        createHomeMenu(requireContext(), WeakReference(binding.menuButton))
+        createTabCounterMenu()
+
+        binding.menuButton.setColorFilter(
+            ContextCompat.getColor(
+                requireContext(),
+                ThemeManager.resolveAttribute(R.attr.textPrimary, requireContext())
+            )
+        )
+
+        binding.toolbar.compoundDrawablePadding =
+            view.resources.getDimensionPixelSize(R.dimen.search_bar_search_engine_icon_padding)
+        binding.toolbarWrapper.setOnClickListener {
+            navigateToSearch()
+        }
+
+        binding.tabButton.setOnClickListener {
+            openTabsTray()
+        }
+
+        consumeFrom(onboardingStore) { state->
+            onboardingView.update(state)
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+
+        _binding = null
+    }
+
+    private fun observeSearchEngineChanges() {
```

Comment:
OK, I agree with your point that landing this as-is will cause too much code duplication, even in the short to mid-term. We could follow a similar pattern we use for browser fragment and use an abstract base class here to share the common functionality. Arguably the two versions of the home screen are different modifications based on the state so inheritance wouldn't be bad. Alternatively, we can move this functions to a separate type and make it re-usable that way.